### PR TITLE
Bruce/celebrate with user 1

### PR DESCRIPTION
**Pull request by brucehorn**
Some improvements in the output during edgectl install, working on the Edgectl Install Improvements story, "Celebrate with the User".  I made a first pass on the following (from the story):

-_change the edgectl install success message printed to the terminal to a happier, more celebratory success message: bold? Capital letters? Distinguished spacing? This seems silly, but it turns out that a little design goes a long way in helping people like the product enough to put more time into evaluating it._. I colorized and restyled the text a bit.

- _add to the terminal message, some information about how to log back into the UI in the future._  Text was added for this.

-_change the edgectl --help message about login to be more clear that this is how one opens the UI._ Very minor change in the Cobra command definition.

I would appreciate some feedback on the language here.

celebrate2

### CODE REVIEW

**Comment by brucehorn** [comment]
I noticed that I capitalized "To access" which is actually mid-sentence.  Will fix that and am looking forward to other feedback as well.

File: docs/docs/latest/tutorials/getting-started.md [hunk 1]
```diff
@@ -25,15 +25,15 @@ Windows. For other options, such as Docker, read the [detailed instructions] (/d
...
-    sudo curl -fL https://metriton.datawire.io/downloads/darwin/edgectl -o /usr/local/bin/edgectl && sudo chmod a+X /usr/local/bin/edgectl
+    sudo curl -fL https://metriton.datawire.io/downloads/darwin/edgectl -o /usr/local/bin/edgectl && sudo chmod a+x /usr/local/bin/edgectl
     ```
 
     If you decide to download the file, you may encounter a security block. To change this:
     * Go to **System Preferences > Security & Privacy > General**.
     * Click the **Open Anyway** button.
     * On the new dialog, click the **Open** button.
 
-2. Run the installer with `./edgectl install`
```
Comment (bnfb) [inline docs/docs/latest/tutorials/getting-started.md]: Why did you remove the ./ ? Many people's PATH does not include . and thus one has to use ./edgectl rather than just edgectl in order to execute it.

File: docs/docs/latest/tutorials/getting-started.md [hunk 2]
```diff
@@ -63,9 +63,9 @@ Windows. For other options, such as Docker, read the [detailed instructions] (/d
 
 </summary>
 
-1. Download the `edgectl` file
+1. Download the `edgectl.exe` file
    [here](https://metriton.datawire.io/downloads/windows/edgectl.exe).
-2. Run the installer with `edgectl.exe install`
```
Comment (bnfb) [inline docs/docs/latest/tutorials/getting-started.md]: This seems reasonable but I forget how Windows works: can you run a program without the exe on the end?

File: cmd/edgectl/aes_install.go [hunk 3]
```diff
@@ -80,6 +85,16 @@ func aesInstall(cmd *cobra.Command, args []string) error {
 	kcontext, _ := cmd.Flags().GetString("context")
 	i := NewInstaller(verbose)
 
+	// If Scout is disabled (environment variable set to non-null), inform the user.
+	// TODO: should be i.scout.Disabled() when Alex's PR #2496 is merged.
+	if os.Getenv("SCOUT_DISABLE") != "" {
+		i.show.Printf(color.Notice.Sprintf("[aesInstall] Scout is disabled; metrics will not be written to Metriton."))
```
Comment (bnfb) [inline cmd/edgectl/aes_install.go]: This message is not going to be clear to the user (in the odd case that they use it) and thus it will cause them to be confused. Something like: "[aesInstall] Info: phone-home metrics have been disabled by environment variable." Why is this better? (a) it says "Info" so that users know that they can ignore it; (b) it explains a little bit about what is going on in language the user probably understands rather than Scout and Metriton".

File: cmd/edgectl/aes_install.go [hunk 4]
```diff
@@ -80,6 +85,16 @@ func aesInstall(cmd *cobra.Command, args []string) error {
...
+	}
+
+	// Both printed and logged when verbose.
+	i.log.Printf("[aesInstall] the install_id is: %s", i.scout.installID)
```
Comment (bnfb) [inline cmd/edgectl/aes_install.go]: Same comment here: the language should be user-centric. Something like: "[aesInstall] Info: unique id for this install: %s, %s". And it should only print this if Scout is enabled. And it should be just one line (one line is less "it must be important" to the user than two lines).

File: cmd/edgectl/aes_install.go (hunk 3 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: thanks, that's helpful.  I was under the impression this was a "datawire only" message so it is clear that this is better.

File: cmd/edgectl/aes_install.go (hunk 4 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: Sounds good.

File: docs/docs/latest/tutorials/getting-started.md (hunk 1 above)
Comment (brucehorn) [reply docs/docs/latest/tutorials/getting-started.md]: this is some of the code that I somehow inherited--I didn't make those changes, so wanted to figure out how to separate out my changes from the others.

**Comment by ark3** [comment]
This PR shows lots of unrelated commits because the "base" branch is master. Please edit the PR in the GitHub UI to set the base branch to release/v1.3.2.

File: cmd/edgectl/aes_install.go [hunk 5]
```diff
@@ -612,7 +629,7 @@ func (i *Installer) Perform(kcontext string) error {
 		return err
 	}
 	i.Report("cluster_accessible")
-	i.show.Println("Your AES installation's address is", i.address)
+	i.show.Println("Your AES installation's address is", chalk.Bold.TextStyle(i.address))
```
Comment (ark3) [inline cmd/edgectl/aes_install.go]: Can we drop `chalk` and use `gookit/color`  for everything? These two dependencies overlap in functionality, and I prefer `gookit/color` because it supports Windows.

File: cmd/edgectl/aes_install.go [hunk 6]
```diff
@@ -987,24 +1009,35 @@ Removing the CRDs will cause your existing Ambassador Mappings and other resourc
...
-Aborting the installer to avoid corrupting an existing (but undetected) installation.
+The installer will now quit to avoid corrupting an existing (but undetected) installation.
 `
 
-const seeDocs = "See https://www.getambassador.io/docs/latest/tutorials/getting-started/"
+const seeDocs    = "See https://www.getambassador.io/docs/latest/tutorials/getting-started/"
+const seeDocsURL = "https://www.getambassador.io/docs/latest/tutorials/getting-started/"
+
+const phoneHomeDisabled  = "[aesInstall] Info: phone-home is disabled by environment variable"
+const installAndTraceIDs = "[aesInstall] Info: install_id = %s; trace_id = %s"
```
Comment (ark3) [inline cmd/edgectl/aes_install.go]: Why the `[aesInstall]` prefix here? We don't use this anywhere else.

File: cmd/edgectl/main.go [hunk 7]
```diff
@@ -183,7 +183,7 @@ func getRootCommand() *cobra.Command {
 	}
 	loginCmd := &cobra.Command{
 		Use:   "login [flags] HOSTNAME",
-		Short: "Access the Ambassador Edge Policy Console",
+		Short: "Log in and open the Ambassador Edge Policy Console web application",
```
Comment (ark3) [inline cmd/edgectl/main.go]: The [marketing-approved documentation](https://www.getambassador.io/about/edge-policy-console/) says 

> After you install the Ambassador Edge Stack, you can log in to the Edge Policy Console (EPC) to manage your deployment.

Drawing on that phrasing, I suggest "Log in to the Ambassador Edge Policy Console"

File: cmd/edgectl/aes_install.go [hunk 8]
```diff
@@ -83,6 +86,15 @@ func aesInstall(cmd *cobra.Command, args []string) error {
 	kcontext, _ := cmd.Flags().GetString("context")
 	i := NewInstaller(verbose)
 
+	// If Scout is disabled (environment variable set to non-null), inform the user.
+	// TODO: should be i.scout.Disabled() when Alex's PR #2496 is merged.
+	if os.Getenv("SCOUT_DISABLE") != "" {
+		i.show.Printf(color.Info.Sprintf(phoneHomeDisabled))
+	}
```
Comment (ark3) [inline cmd/edgectl/aes_install.go]: Now that #2496 has landed on the release branch, you can update this.

File: cmd/edgectl/aes_install.go [hunk 9]
```diff
@@ -735,8 +752,13 @@ func (i *Installer) Perform(kcontext string) error {
...
 	}
 
+	// Add a little color
+
 	i.show.Println()
-	i.ShowWrapped(fmt.Sprintf(fullSuccess, i.hostname))
+	i.ShowWrapped(color.Info.Sprintf(fullSuccess,
+		chalk.Bold.TextStyle(i.hostname),
+		chalk.Bold.TextStyle("edgectl install "+i.hostname)))
+
```
Comment (ark3) [inline cmd/edgectl/aes_install.go]: I think this should be `login` not `install`.

Comment (ark3) [reply cmd/edgectl/aes_install.go]: Oh, and the namespace flag... You have it correct in the comment below.

File: cmd/edgectl/main.go (hunk 7 above)
Comment (ark3) [reply cmd/edgectl/main.go]: In particular, I don't think we should use the phrase "web application" to describe the console. Yes, I know that does describe it correctly, but our documentation consistently uses that phrase to describe a service that a user writes and then exposes using Ambassador.

File: docs/docs/latest/tutorials/getting-started.md (hunk 1 above)
Comment (brucehorn) [reply docs/docs/latest/tutorials/getting-started.md]: this is no longer the case now that the PR is correctly branched off of release/v1.3.2

File: docs/docs/latest/tutorials/getting-started.md (hunk 2 above)
Comment (brucehorn) [reply docs/docs/latest/tutorials/getting-started.md]: this is no longer code in this PR now that I've branched off of release/v1.3.2

File: cmd/edgectl/aes_install.go (hunk 3 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: Fixed in new push.

File: cmd/edgectl/aes_install.go (hunk 5 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: I use both because chalk allows different functionality (e.g. changing a word within a sentence to bold without terminating the style change at that point) and they interoperate well.  I will spend some more time to see if I can rewrite the code to properly work with just one or the other.

File: cmd/edgectl/aes_install.go (hunk 6 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: it was meant to be a parallel construct to the installer.SetMetadatum and Report,  which say:
"[Metrics]..."
but I can remove it of course.

File: cmd/edgectl/aes_install.go [hunk 10]
```diff
@@ -987,24 +1009,35 @@ Removing the CRDs will cause your existing Ambassador Mappings and other resourc
 
 $ kubectl delete crd -l product=aes
 
-Aborting the installer to avoid corrupting an existing (but undetected) installation.
+The installer will now quit to avoid corrupting an existing (but undetected) installation.
 `
 
-const seeDocs = "See https://www.getambassador.io/docs/latest/tutorials/getting-started/"
+const seeDocs    = "See https://www.getambassador.io/docs/latest/tutorials/getting-started/"
```
Comment (khussey) [inline cmd/edgectl/aes_install.go]: Should seeDocs somehow reference or build on seeDocsURL (so the URL string doesn't exist in two places)?

File: cmd/edgectl/main.go (hunk 7 above)
Comment (brucehorn) [reply cmd/edgectl/main.go]: The point of the word "open" is that it describes the fact that it will open in the web browser; logging in is just the beginning of the process.

Removing the "web application" text sounds right given the need to stay consistent with the documentation.

File: cmd/edgectl/aes_install.go (hunk 8 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: will do

File: cmd/edgectl/aes_install.go (hunk 10 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: I was thinking about that too.  I'll see if Go will let me construct a const from string fragments, otherwise I'll have to pass it in as a parameter.

File: cmd/edgectl/aes_install.go (hunk 8 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: done

File: cmd/edgectl/aes_install.go (hunk 5 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: Windows support is a requirement. If `chalk` doesn't _break_ Windows (and just leaves things unstyled in that case), then I suppose that's okay. Also, it looks like `chalk` might be unmaintained.

File: cmd/edgectl/aes_install.go (hunk 9 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: I wasn't sure since reading elsewhere in the code says that `--namespace ambassador` is the default.  Do we want to always have our users specify the namespace?

File: cmd/edgectl/aes_install.go (hunk 6 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: The `[Metrics]` prefix et al are not seen by the user (except in verbose mode, which is really just "log to the screen" mode -- it could use a lot of refinement).

File: cmd/edgectl/aes_install.go (hunk 9 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: Ah yes, for `edgectl login` the default for the namespace argument works fine. You are right.

File: cmd/edgectl/aes_install.go (hunk 10 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: That feature exists in the code I'm working on. I think we can get Bruce's PR into 1.3.2, whereas my work will take longer. Let's leave in the duplication for now and improve things in the next release.

**Comment by brucehorn** [comment]
Latest install looks like this:

Screen Shot 2020-03-30 at 12 28 51 PM

File: docs/docs/latest/tutorials/getting-started.md (hunk 1 above)
Comment (khussey) [reply docs/docs/latest/tutorials/getting-started.md]: @bnfb the commands to download and adjust permissions for the binary actually place it in /usr/local/bin so in fact attempting to run the command with a prefix of "./" will fail.

File: docs/docs/latest/tutorials/getting-started.md (hunk 2 above)
Comment (khussey) [reply docs/docs/latest/tutorials/getting-started.md]: @bnfb you can indeed run a program on Windows without specifying the .exe extension.

File: cmd/edgectl/main.go (hunk 7 above)
Comment (khussey) [reply cmd/edgectl/main.go]: @brucehorn did you decide to abandon the "and open" as well (which I thought was an improvement)?

Comment (brucehorn) [reply cmd/edgectl/main.go]: I thought it was an improvement as well but deferred to @ark3

File: cmd/edgectl/aes_install.go (hunk 10 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: fixed already in this PR

**Comment by brucehorn** [comment]
Still waiting on a pending reviewer before merging...

File: cmd/edgectl/aes_install.go (hunk 5 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: latest push resolves this--`chalk` is no longer a requirement.

File: cmd/edgectl/aes_install.go (hunk 9 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: resolved, fixed `login` and removed the namespace text.

File: cmd/edgectl/aes_install.go (hunk 6 above)
Comment (brucehorn) [reply cmd/edgectl/aes_install.go]: Removed in this PR.
